### PR TITLE
373 competing interest

### PR DESCRIPTION
**Pull request by ilya-zhuravel**

### CODE REVIEW

File: src/app/containers/author-form-dialog/index.tsx [hunk 1]
```diff
@@ -155,6 +165,34 @@ export const AuthorFormDialog: React.FC<AuthorFormDialogProps> = (props) => {
         value={author.email || ''}
         onChange={handleFormChange}
       />
+      <FormControl variant="outlined" classes={{ root: classes.inputField }}>
+        <InputLabel shrink id="author-affiliations-label">
+          Affiliated Author
```
Comment (NuclearRedeye) [inline src/app/containers/author-form-dialog/index.tsx:170]: Wrong Label, should be 'Competing interest'

File: src/app/containers/author-form-dialog/index.tsx [hunk 2]
```diff
@@ -155,6 +165,34 @@ export const AuthorFormDialog: React.FC<AuthorFormDialogProps> = (props) => {
         value={author.email || ''}
         onChange={handleFormChange}
       />
+      <FormControl variant="outlined" classes={{ root: classes.inputField }}>
+        <InputLabel shrink id="author-affiliations-label">
```
Comment (NuclearRedeye) [inline src/app/containers/author-form-dialog/index.tsx:169]: Design shows the control as being full width, hence maybe the shrink shouldn't be here (if that's what shrink even does)?

File: src/app/models/person.ts [hunk 3]
```diff
@@ -47,19 +49,29 @@ export function getAuthorAffiliationsLabels(author: Person, affiliations: Affili
...
       suffix: getTextContentFromPath(author, 'name > suffix'),
       bio: createBioEditorState(author.querySelector('bio')),
       email: getTextContentFromPath(author, 'email'),
       orcId: getTextContentFromPath(author, 'contrib-id[contrib-id-type="orcid"]'),
       isCorrespondingAuthor: author.getAttribute('corresp') === 'yes',
-      affiliations: Array.from(author.querySelectorAll('xref[ref-type="aff"]')).map((xRef) => xRef.getAttribute('rid'))
-    })
-  );
+      affiliations: Array.from(author.querySelectorAll('xref[ref-type="aff"]')).map((xRef) => xRef.getAttribute('rid')),
+      hasCompetingInterest: Boolean(competingInterests),
```
Comment (NuclearRedeye) [inline src/app/models/person.ts:71]: This needs to be more clever, as in the input XML there technically is a competing interest for each author although in the majority of cases that value is always 'No competing interests declared', as such I think something like...
```
hasCompetingInterest(text) {
    return text && text !== "No competing interests declared";
}
```

File: src/app/containers/manuscript/authors-info-details/index.tsx [hunk 4]
```diff
@@ -77,7 +77,11 @@ const AuthorInfo: React.FC<AuthorInfoProps> = React.memo(({ author }) => {
         <div className={classes.authorInfoLine}>
           <img src={OrcidIcon} alt="orcid icon" className={classes.orcidIcon} /> 0000-0001-5000-0007
         </div>
-        <div className={classes.authorInfoLine}> No Competing interest declared </div>
+        <div className={classes.authorInfoLine}>
+          {author.hasCompetingInterest
+            ? `Competing interest: ${author.competingInterestStatement}`
```
Comment (NuclearRedeye) [inline src/app/containers/manuscript/authors-info-details/index.tsx:82]: As a result of the issue noted below, I see...
Screenshot 2020-07-06 at 13 31 56
But this should resolve if you sort out the 'hasCompetingInterest' flag.

File: src/app/containers/author-form-dialog/index.tsx (hunk 1 above)
Comment (ilya-zhuravel) [reply src/app/containers/author-form-dialog/index.tsx]: image

File: src/app/containers/author-form-dialog/index.tsx (hunk 2 above)
Comment (ilya-zhuravel) [reply src/app/containers/author-form-dialog/index.tsx:169]: Not really, the shrink is for the label to sit at the top. the fullWidth attribute does the trick.

File: src/app/models/person.ts (hunk 3 above)
Comment (ilya-zhuravel) [reply src/app/models/person.ts]: Yeah I thought it looked rather dumb but assumed the xml isn't right.

File: src/app/containers/manuscript/authors-info-details/index.tsx [hunk 5]
```diff
@@ -77,7 +77,11 @@ const AuthorInfo: React.FC<AuthorInfoProps> = React.memo(({ author }) => {
...
+            : 'No Competing interest declared'}
```
Comment (NuclearRedeye) [inline src/app/containers/manuscript/authors-info-details/index.tsx:83]: Small typo, should be 'No competing interests declared', as that is what is in the XML.

File: src/app/models/person.ts (hunk 3 above)
Comment (NuclearRedeye) [reply src/app/models/person.ts]: I makes sense from the XML perspective, but yeah it's quirky to handle when loading the content.

**Pull request merged**